src/layout/strategy/collision.js

```diff
@@ -0,0 +1,88 @@
+export default function() {
+
+    var data = [];
+    var collision = function(inputData) {
+        data = inputData;
+
+        return collision;
+    };
+
+    collision.getAllCollisions = function(index) {
+        var allCollisions;
+
+        if (!arguments.length) {
+            allCollisions = {};
+            collideAll(function(a, b) {
+                if (allCollisions[a] === undefined) {
+                    allCollisions[a] = [];
+                }
+
+                allCollisions[a].push(b);
+            });
+        } else {
+            allCollisions = [];
+            collidePoint(index, function(a, b) {
+                allCollisions.push(b);
+            });
+        }
+
+        return allCollisions;
+    };
+
+    collision.numberOfCollisions = function(index) {
+        var collisions = 0;
+
+        if (!arguments.length) {
+            collideAll(function(a, b) {
+                collisions++;
+            });
+        } else {
+            collidePoint(index, function(a, b) {
+                collisions++;
+            });
+        }
+
+        return collisions;
+    };
+
+    collision.anyCollisions = function(index) {
+        var collisions = false;
+
+        if (!arguments.length) {
+            collideAll(function() {
+                collisions = true;
+            });
+        } else {
+            collidePoint(index, function() {
+                collisions = true;
+            });
+        }
+
+        return collisions;
+    };
+
+    function isIntersecting(a, b) {
+        return !(a.x > (b.x + b.width) ||
+            (a.x + a.width) < b.x ||
+            a.y > (b.y + b.height) ||
+            (a.y + a.height) < b.y);
+    }
+
+    function collideAll(callback) {
+        for (var i = 0; i < data.length; i++) {
+            collidePoint(i, callback);
+        }
+    }
+
+    function collidePoint(pointIndex, callback) {
```

Comment:
Rather than having a 'visitor' callback, how about a more functional approach ... where as an input you have an array of data, then use Array.filter to output the points that collide?